Gordon360/Services/RecIM/TeamService.cs

```diff
@@ -357,47 +355,50 @@ public async Task<TeamViewModel> UpdateTeamAsync(int teamID, TeamPatchViewModel
             t.Name = update.Name ?? t.Name;
             t.StatusID = update.StatusID ?? t.StatusID;
             
-            if (update.Logo != null)
+            if (update.IsLogoUpdate)
             {
-                // ImageUtils.GetImageFormat checks whether the image type is valid (jpg/jpeg/png)
-                var (extension, format, data) = ImageUtils.GetImageFormat(update.Logo);
-
-                string? imagePath = null;
-                // If old image exists, overwrite it with new image at same path
-                if (t.Logo != null)
+                if (update.Logo != null)
                 {
-                    imagePath = GetImagePath(Path.GetFileName(t.Logo));
-                }
-                // Otherwise, upload new image and save url to db
-                else
-                {
-                    // Use a unique alphanumeric GUID string as the file name
-                    var filename = $"{Guid.NewGuid().ToString("N")}.{extension}";
-                    imagePath = GetImagePath(filename);
-                    var url = GetImageURL(filename);
-                    t.Logo = url;
-                }
+                    // ImageUtils.GetImageFormat checks whether the image type is valid (jpg/jpeg/png)
+                    var (extension, format, data) = ImageUtils.GetImageFormat(update.Logo);
 
-                ImageUtils.UploadImage(imagePath, data, format);
-            }
+                    string? imagePath = null;
+                    // If old image exists, overwrite it with new image at same path
+                    if (t.Logo != null)
+                    {
+                        imagePath = GetImagePath(Path.GetFileName(t.Logo));
+                    }
+                    // Otherwise, upload new image and save url to db
+                    else
+                    {
+                        // Use a unique alphanumeric GUID string as the file name
+                        var filename = $"{Guid.NewGuid().ToString("N")}.{extension}";
+                        imagePath = GetImagePath(filename);
+                        var url = GetImageURL(filename);
+                        t.Logo = url;
+                    }
+
+                    ImageUtils.UploadImage(imagePath, data, format);
+                }
```

Comment:
Same as above, a lot of repeated code. Factor out?

Reply:
I'll take a look at it. @Kevin-Chu58 if you get to it before me, feel free to make the adjustments.